pkg/asset/machines/master.go

```diff
@@ -172,6 +172,23 @@ func (m *Master) Generate(ctx context.Context, dependencies asset.Parents) error
 	var ipClaims []ipamv1.IPAddressClaim
 	var ipAddrs []ipamv1.IPAddress
 	var controlPlaneMachineSet *machinev1.ControlPlaneMachineSet
+
+	// Check if SNO topology is supported on this platform
+	if pool.Replicas != nil && *pool.Replicas == 1 {
+		bootstrapInPlace := false
+		if ic.BootstrapInPlace != nil {
+			bootstrapInPlace = true
+		}
+		if !supportedSingleNodePlatform(bootstrapInPlace, ic.Platform.Name()) {
+			return errors.New("This install method does not support Single Node installation on this platform")
+		}
+		/* Is this test required ?
+		mpool := ic.WorkerMachinePool()
+		if mpool != nil && mpool.Replicas != ptr.To(int64(0)) {
```

Comment:
When there is a single master node, is there some restriction around only supporting 0 worker nodes? I know in ABI that is the case, I am not sure about IPI. @zaneb could you comment?

Reply:
On platform none/external, it is not possible to have a day 1 worker with SNO. But on cloud platforms I assume it is ok as the MachineSet on the control plane will create the worker.